### DISCUSSION TITLE
Send the results back to Emacs in a dired/grep buffer

### DISCUSSION THREAD

**Discussion by maxecharel** [eval-exec/eee.el#21] [Ideas]
Hi @eval-exec! Thanks a lot (again) for this package; as previously mentioned on Reddit and in a previous issue, it is super useful for externalizing some demanding find/grep tasks that would possibly put too much burden on Emacs thread (hello ripgrep-all).
If I may, I would like to kindly ask for an additional functionality: being able to send back to Emacs the results of a search, as a dired/grep buffer. Do you think that something you could (and want to) implement?

**Comment by eval-exec** [comment]
Hello, @maxecharel 
Do you mean something like this?

Starting eee.el in Emacs, which in turn launches a third-party terminal emulator. Inside the terminal emulator, it runs a command similar to ripgrep or find, then sends the command output back to Emacs, displaying the results in a buffer.

I think this is achievable.

It would likely require defining a custom callback function. In the callback, Emacs could create a buffer and insert the command output from eee.el into that buffer:
https://github.com/eval-exec/eee.el/blob/dce7135c446d990169c6e119eb780d20078bb4d9/eee.el#L194-L206

**Comment by eval-exec** [comment]
> Do you think that something you could (and want to) implement?

Yes, for example:
```elisp
;; write a custom callback function to create a buffer in Emacs, and write something into the buffer:
(defun ee-export-to-buffer (destination-file)
  "Export the contents of DESTINATION-FILE to a new buffer."
  (let* ((destination (shell-command-to-string
						(format "cat %s" destination-file)))
		  (destination (string-trim destination)))
	;; Create a new buffer and insert the content
	(with-current-buffer (get-buffer-create "*Export Output*")
	  (erase-buffer)
	  (insert destination)
	  (goto-char (point-min))
	  (display-buffer (current-buffer)))))

;; then use ee-define to create a new "ee" command:
(ee-define "ee-rg-and-export-to-buffer" default-directory (ee-script-path "eee-rg.sh") nil ee-export-to-buffer)
```
Note: I weite above code by hand, not tested it. you may need to test it. Thank you.

**Comment by maxecharel** [comment]
Waouw, thanks for the reactivity! I tested the code (just added a call to `grep-mode` at the end) and yes, that is the spirit. But I was more specifically thinking about a way to export *all* matches to a grep (or dired, if using `find` or `fd`) buffer instead of just the match at point in the third-party terminal emulator, in the spirit of what I get when calling `empark-export` from e.g. `consult-ripgrep`. Do you think that would induce a substantial additional effort?

**Reply by maxecharel** [reply]
(In other words, to be able to do exactly what the combination `consult-ripgrep` -- that shows the matches in the minibuffer -- and `embark-export` -- that exports *all* the matches to a grep buffer -- does, but having the search and export performed via `eee.el` instead of respectively `consult-ripgrep` and `embark-export`)
